src/vs/workbench/contrib/terminal/browser/media/fish_xdg_data/fish/vendor_conf.d/shellIntegration.fish

```diff
@@ -98,6 +98,32 @@ function __vsc_cmd_clear --on-event fish_cancel
 	__vsc_esc D
 end
 
+# Preserve the user's existing prompt, to wrap in our escape sequences.
+function __preserve_fish_prompt --on-event fish_prompt
+	if functions --query fish_prompt
+		if functions --query __vsc_fish_prompt
+			# Erase the fallback so it can be set to the user's prompt
+			functions --erase __vsc_fish_prompt
+		end
+		functions --copy fish_prompt __vsc_fish_prompt
+		functions --erase __preserve_fish_prompt
+		# Now __vsc_fish_prompt is guaranteed to be defined
+		__init_vscode_shell_integration
+	else
+		if functions --query __vsc_fish_prompt
+			# There is no fish_prompt set, so stick with the default
```

Comment:
This comment reads to me like it belongs to the inner else branch (line 119, where `__vsc_fish_prompt` is set to the default prompt)